README.md

```diff
@@ -45,20 +56,69 @@ You can either keep running the code from source, using `cargo run --release --b
 
+If the `RUST_LOG` environment variable is not specified, the log level `info` will be used by default.
+
 Here are the commands as recommended for different scenarios:
 
+#### Assuming you're using the source code
+
+Every release has a tag and a release branch. Make sure you checkout the release you need. For example, if you need v0.3.0, you should first run the following to check out the release branch:
+
+```sh
+git checkout release-v0.3.0
+```
+
+or to checkout the tag:
+
+```sh
+git checkout tags/v0.3.0
+```
+
+Release branches are more recommended than tags, because they get necessary security patches, if any. Tags are just markers.
+
+After having checked out the version you need, the following is how to run the software.
+
 For normal operation (replace testnet by mainnet to run the mainnet node)
+You don't need to specify RUST_LOG environment variable for logging, since it's by default set to be `INFO`.
+
 Note: spaces don't matter, so these are aligned for readability
-- Node daemon: `RUST_LOG=info cargo run --release --bin node-daemon       -- testnet 2>&1 | tee ../mintlayer.log`
-- CLI Wallet:  `RUST_LOG=info cargo run --release --bin wallet-cli        -- testnet 2>&1 | tee ../wallet-cli.log`
-- RPC Wallet:  `RUST_LOG=info cargo run --release --bin wallet-rpc-daemon -- testnet 2>&1 | tee ../wallet-cli.log`
-- GUI:         `RUST_LOG=info cargo run --release --bin node-gui          -- testnet 2>&1 | tee ../node-gui.log`
+
+- Node daemon:
+```sh
+cargo run --release --bin node-daemon       -- testnet 2>&1 | tee ../mintlayer.log
+```
+- CLI Wallet:
+```sh
+cargo run --release --bin wallet-cli        -- testnet 2>&1 | tee ../wallet-cli.log
+```
+- RPC Wallet:
+```sh
+cargo run --release --bin wallet-rpc-daemon -- testnet 2>&1 | tee ../wallet-cli.log
+```
+- GUI:
+```sh
+cargo run --release --bin node-gui          -- testnet 2>&1 | tee ../node-gui.log
+```
 
 For heavy debugging operation
-- Node daemon: `RUST_LOG=debug cargo run --bin node-daemon       -- testnet 2>&1 | tee ../mintlayer.log`
-- CLI Wallet:  `RUST_LOG=debug cargo run --bin wallet-cli        -- testnet 2>&1 | tee ../wallet-cli.log`
-- RPC Wallet:  `RUST_LOG=debug cargo run --bin wallet-rpc-daemon -- testnet 2>&1 | tee ../wallet-cli.log`
-- GUI:         `RUST_LOG=debug cargo run --bin node-gui          -- testnet 2>&1 | tee ../node-gui.log`
+- Node daemon:
+```sh
+RUST_LOG=debug cargo run --bin node-daemon       -- testnet 2>&1 | tee ../mintlayer.log
+```
+- CLI Wallet:
+```sh
+RUST_LOG=debug cargo run --bin wallet-cli        -- testnet 2>&1 | tee ../wallet-cli.log
+```
+- RPC Wallet:
+```sh
+RUST_LOG=debug cargo run --bin wallet-rpc-daemon -- testnet 2>&1 | tee ../wallet-cli.log
+```
+- GUI:
+```sh
+RUST_LOG=debug cargo run --bin node-gui          -- testnet 2>&1 | tee ../node-gui.log
+```
+
+There are more complex logging filtering that can be done at the crate level, but this is out of scope.
 
```

Comment:
`are` and `filtering` don't add up. I'd rephrase this, e.g. as follows:

> More complex log filtering is possible, e.g. you can filter out log lines produced by a certain crate or module, but this is out of scope of this document.

Reply:
Done. Thanks.